lib/AST/DiagnosticEngine.cpp

```diff
@@ -577,7 +577,21 @@ static void formatDiagnosticArgument(StringRef Modifier,
       if (shouldShowAKA(type, typeName)) {
         llvm::SmallString<256> AkaText;
         llvm::raw_svector_ostream OutAka(AkaText);
-        OutAka << type->getCanonicalType();
+
+        Type desugaredType;
+        // For aka generic function types just desugar aliases on
+        // the type.
+        if (type->is<GenericFunctionType>()) {
+          desugaredType = type.transform([](Type visitTy) -> Type {
+            if (isa<TypeAliasType>(visitTy.getPointer()))
+              return visitTy->getDesugaredType();
```

Comment:
Can you factor this out into a function and recursively desugar the desugared type here? Otherwise, its won't work if the type alias has components that are type aliases, eg

```
typealias Foo = Int
typealias Bar = (Foo) -> ()
```

Reply:
Fixed!